goyaml.v3/emitterc.go

```diff
@@ -243,6 +243,9 @@ func yaml_emitter_increase_indent(emitter *yaml_emitter_t, flow, indentless bool
 			// Everything else aligns to the chosen indentation.
 			emitter.indent = emitter.best_indent * ((emitter.indent + emitter.best_indent) / emitter.best_indent)
 		}
+		if compact_seq {
+			emitter.indent = emitter.indent - 2
```

Comment:
Yeah, IIRC it needs to subtract 2 in both situations. 

The value `compact_seq` passed in is always set to false when this function is called, except when we are dealing with sequence nodes. So this gets triggered to subtract 2 only when we are increasing the indent to account for sequence nodes, which will be correct because we need to subtract 2 to account for the `- ` at the beginning of the sequence node. 

I will add this as a comment.